pkg/didcomm/dispatcher/dispatcher.go

```diff
@@ -15,6 +15,7 @@ type Service interface {
 
 // DIDCommMsg did comm msg
 type DIDCommMsg struct {
-	Type    string
-	Payload []byte
+	Outbound bool
```

Comment:
Godoc would be helpful as what value need to be set here.

Reply:
done

Reply:
I think we don't need Outbound bool flag as from the msgtype you would know the state and follow up state

Reply:
@fqutishat this comment illustrates why it's useful to know the direction the message is heading in: https://github.com/hyperledger/aries-framework-go/issues/231#issuecomment-529939840

To be fair - what we *actually* need to know is what role we're participating in in this protocol. This PR uses the message's direction as a proxy for that. But maybe it's worthwhile to look into how to just store our role (as well as state) instead of changing `dispatcher.DIDCommMsg`. I'll look into this in a couple of hours.

Reply:
@fqutishat I think it's *crucial* to know the message's direction.

Look again at this comment: https://github.com/hyperledger/aries-framework-go/issues/231#issuecomment-529939840

Now let's ask ourselves: `Service.Handle(msg)` just received a `Request` message. It doesn't know whether it's inbound or outbound. It doesn't have any context persisted to the store, so okay - it knows this is a new instance of the did-exchange protocol. What should it do now - should it just call the listeners, or should it send the `Request` message to the other agent?